addons/netfox.extras/state-machine/rewindable-state-machine.gd

```diff
@@ -58,20 +58,26 @@ func transition(new_state_name: StringName) -> void:
 		
 	var new_state: RewindableState = _available_states[new_state_name]
 	if _state_object:
-		if !new_state.can_enter(_state_object):
+		if !is_rollback and !new_state.can_enter(_state_object):
 			return
 	
-		_state_object.exit(new_state, NetworkRollback.tick)
+		_state_object.exit(new_state, NetworkRollback.tick, is_rollback)
 	
 	var _previous_state: RewindableState = _state_object
 	_state_object = new_state
-	on_state_changed.emit(_previous_state, new_state)
-	_state_object.enter(_previous_state, NetworkRollback.tick)
+	_state_object.enter(_previous_state, NetworkRollback.tick, is_rollback)
 
-func _ready():
-	# Gather known states
-	for child in find_children("*", "RewindableState", false):
-		_available_states[child.name] = child
+func _notification(what: int):
+	if what == NOTIFICATION_READY:
```

Comment:
This does the same as the `_ready` callback, why the change?

Reply:
it's so users can comfortably add their own logic without overriding what's underneath (which is what batteryaciddev bumped into). I know you can call _ready.super() but this is just an easier solution.

Reply:
Gotcha! I've restored this form. Though long-term I'm unsure of this pattern - while this is comfortable for the user, it's less explicit and more magic. 

Anyway, let's leave this in and see the reactions.